scanpy/tests/test_plotting.py

```diff
@@ -1215,6 +1215,57 @@ def test_scatter_specify_layer_and_raw():
         sc.pl.umap(pbmc, color="HES4", use_raw=True, layer="layer")
 
 
+@pytest.fixture
+def pbmc_filtered():
+    pbmc = sc.datasets.pbmc68k_reduced()
+    sc.pp.filter_genes(pbmc, min_cells=10)
+    return pbmc
+
+
+@pytest.mark.parametrize(
+    "x,y,color,use_raw,value_error",
+    [
+        # test that plots work with var_names only found in raw if use_raw
+        # is None or True
+        ('EGFL7', 'F12', 'FAM185A', None, False),
+        ('EGFL7', 'F12', 'FAM185A', True, False),
+        # test that plotting fails with a ValueError if trying to plot
+        # var_names only found in raw and use_raw is False
+        ('EGFL7', 'F12', 'FAM185A', False, True),
+        # test that plotting one variable from var.index vs. another from
+        # obs.keys() works, regardless of use_raw
+        ('HES4', 'percent_mito', None, False, False),
+        ('HES4', 'percent_mito', None, True, False),
+        # test that plotting fails if one axis is a per-var value and the
+        # other is a per-obs value
+        ('HES4', 'n_cells', None, None, True),
+        ('percent_mito', 'AAAGCCTGGCTAAC-1', None, None, True),
+        # test that plotting one variable from obs.index vs. another from
+        # var.keys() works, regardless of use_raw
```

Comment:
Should this work regardless of `use_raw`?

`pbmc` doesn't have `.var["n_cells"]`, so this should fail with a key / value error for the commented out case, no?

Reply:
It does after the filtering step included in the `pbmc_filtered` fixture.

Reply:
Ah, gotcha